sql/postgresql/PostgreSQLParser.g4

```diff
@@ -3583,8 +3445,8 @@ xml_namespace_el
    ;
 
 typename
-   : simpletypename (opt_array_bounds |ARRAY (OPEN_BRACKET iconst CLOSE_BRACKET)?)
-   | SETOF simpletypename (opt_array_bounds |ARRAY (OPEN_BRACKET iconst CLOSE_BRACKET)?)
+   : simpletypename (opt_array_bounds | ARRAY (OPEN_BRACKET iconst CLOSE_BRACKET)?)
+   | SETOF simpletypename (opt_array_bounds | ARRAY (OPEN_BRACKET iconst CLOSE_BRACKET)?)
```

Comment:
```suggestion
   : SETOF? simpletypename (opt_array_bounds | ARRAY (OPEN_BRACKET iconst CLOSE_BRACKET)?)
```

Reply:
the same as above - avoid leading optional values